lib/mergeBuffers.js

```diff
@@ -23,12 +24,29 @@ function mergeBuffers(gltf, defaultName) {
     });
     baseBufferName = defaultValue(baseBufferName, 'buffer');
 
+    let buffersByteLength = 0;
+    ForEach.buffer(gltf, function(buffer) {
+        buffersByteLength += buffer.extras._pipeline.source.length;
+    });
+
+    // Don't merge buffers if the merged buffer will exceed the Node limit.
+    const splitBuffers = (buffersByteLength > mergeBuffers._getBufferMaxByteLength());
+
     const buffersToMerge = {};
+    const mergedNameCount = {};
 
     ForEach.bufferView(gltf, function(bufferView) {
         const buffer = gltf.buffers[bufferView.buffer];
         let mergedName = buffer.extras._pipeline.mergedBufferName;
         mergedName = defined(mergedName) ? baseBufferName + '-' + mergedName : baseBufferName;
+
+        if (splitBuffers) {
+            if (!defined(mergedNameCount[mergedName])) {
+                mergedNameCount[mergedName] = 0;
+            }
+            mergedName += '-' + mergedNameCount[mergedName]++;
+        }
```

Comment:
Looks like this block is ensuring that each buffer gets its own entry in `buffersToMerge` instead of getting lumped with other buffers.

I think `Buffer.concat` as used below will return a buffer referencing a separate chunk of memory even when called on an array of a single buffer. If we're worrying about fairly large buffers (such that concatenating would blow the buffer limit), could the `concat` below only happen when there's more than one buffer?

Reply:
That's a good idea. Fixed.